SignalServiceKit/src/Messages/Attachments/TSAttachmentStream.m

```diff
@@ -98,6 +96,12 @@ - (nullable instancetype)initWithCoder:(NSCoder *)coder
     return self;
 }
 
+- (void)saveWithTransaction:(YapDatabaseReadWriteTransaction *)transaction
+{
+    [super saveWithTransaction:transaction];
+    [self ensureThumbnail];
```

Comment:
But:

* The "convert all at startup" logic only reads and doesn't save.
* For legacy attachments not yet fixed by "convert all at startup", we want to ensure a thumbnail before displaying them.

There's more work to do here, right?

Reply:
initWithCoder still calls ensureThumbnail, just not initWithContentType nor initWithPointer, so it will be called for existing attachments.

Reply:
Sorry for the misreading.